# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/cookie/CookieStorage.cpp
# Commit: 6379a4984eba
# Full Hash: 6379a4984ebafb659428ec3e4493f72173b3a70a
# Author: Paul Zuehlcke <pbz@mozilla.com>
# Date: 2023-09-22 04:22:03
# Description:
#   Bug 1854028 - Update BounceTrackingStorageObserver cookie check and remove nsICookieNotification isThirdPartyCookie. r=bvandersloot,cookie-reviewers,anti-tracking-reviewers,timhuang
#   
#   For the BounceTrackingStorageObserver, instead of passing in a third-party flag
#   we can simply check if the changed cookie is partitioned or not.
#   
# ==============================================================================

diff -r 24c5f04d3793 -r 6379a4984eba netwerk/cookie/CookieStorage.cpp
--- a/netwerk/cookie/CookieStorage.cpp	Thu Sep 21 14:30:17 2023 +0000
+++ b/netwerk/cookie/CookieStorage.cpp	Thu Sep 21 14:42:32 2023 +0000
@@ -373,7 +373,6 @@
                                   nsICookieNotification::Action aAction,
                                   const nsACString& aBaseDomain,
                                   dom::BrowsingContext* aBrowsingContext,
-                                  bool aIsThirdPartyCookie,
                                   bool aOldCookieIsSession) {
   nsCOMPtr<nsIObserverService> os = services::GetObserverService();
   if (!os) {
@@ -394,9 +393,8 @@
     browsingContextId = aBrowsingContext->Id();
   }
 
-  nsCOMPtr<nsICookieNotification> notification =
-      new CookieNotification(aAction, cookie, aBaseDomain, batchDeletedCookies,
-                             browsingContextId, aIsThirdPartyCookie);
+  nsCOMPtr<nsICookieNotification> notification = new CookieNotification(
+      aAction, cookie, aBaseDomain, batchDeletedCookies, browsingContextId);
   // Notify for topic "private-cookie-changed" or "cookie-changed"
   os->NotifyObservers(notification, NotificationTopic(), u"");
 
@@ -414,8 +412,7 @@
                               Cookie* aCookie, int64_t aCurrentTimeInUsec,
                               nsIURI* aHostURI, const nsACString& aCookieHeader,
                               bool aFromHttp,
-                              dom::BrowsingContext* aBrowsingContext,
-                              bool aIsThirdPartyCookie) {
+                              dom::BrowsingContext* aBrowsingContext) {
   int64_t currentTime = aCurrentTimeInUsec / PR_USEC_PER_SEC;
 
   CookieListIter exactIter{};
@@ -536,8 +533,7 @@
         COOKIE_LOGFAILURE(SET_COOKIE, aHostURI, aCookieHeader,
                           "previously stored cookie was deleted");
         NotifyChanged(oldCookie, nsICookieNotification::COOKIE_DELETED,
-                      aBaseDomain, aBrowsingContext, aIsThirdPartyCookie,
-                      oldCookieIsSession);
+                      aBaseDomain, aBrowsingContext, oldCookieIsSession);
         return;
       }
 
@@ -622,8 +618,7 @@
   NotifyChanged(aCookie,
                 foundCookie ? nsICookieNotification::COOKIE_CHANGED
                             : nsICookieNotification::COOKIE_ADDED,
-                aBaseDomain, aBrowsingContext, aIsThirdPartyCookie,
-                oldCookieIsSession);
+                aBaseDomain, aBrowsingContext, oldCookieIsSession);
 }
 
 void CookieStorage::UpdateCookieOldestTime(Cookie* aCookie) {